processor/probabilisticsamplerprocessor/sampler_mode.go

```diff
@@ -28,23 +38,45 @@ const (
 	percentageScaleFactor = numHashBuckets / 100.0
 )
 
-// SamplerMode controls the logic used in making a sampling decision.
-// The HashSeed mode is the only mode, presently, and it is also the
-// default mode.
-//
-// TODO: In the future, when OTEP 235 is introduced, there will be two
-// new modes.
+// samplerErrors are conditions reported by the sampler that are somewhat
+// ordinary and should log as info-level.
+type samplerError string
+
+var _ error = samplerError("")
+
+func (s samplerError) Error() string {
+	return string(s)
+}
+
+// SamplerMode determines which of several modes is used for the
+// sampling decision.
 type SamplerMode string
 
 const (
-	HashSeed    SamplerMode = "hash_seed"
-	DefaultMode SamplerMode = HashSeed
-	modeUnset   SamplerMode = ""
+	// HashSeed applies the hash/fnv hash function originally used in this component.
+	HashSeed SamplerMode = "hash_seed"
+
+	// Equalizing uses OpenTelemetry consistent probability
+	// sampling information (OTEP 235), applies an absolute
+	// threshold to equalize incoming sampling probabilities.
+	Equalizing SamplerMode = "equalizing"
+
+	// Proportional uses OpenTelemetry consistent probability
+	// sampling information (OTEP 235), multiplies incoming
+	// sampling probaiblities.
+	Proportional SamplerMode = "proportional"
+
+	// defaultNoHashSeedMode is applied when the mode is unset and
+	// sampling is based on TraceID and not logs attributes.
+	// Proportional is a better default in this case because it
+	// recognizes OTEP 235 sampling inputs and lowers sampling
+	// probability consistently.
+	defaultNoHashSeedMode SamplerMode = Proportional
```

Comment:
What happens if we have one collector with the legacy hash_seed mode, and another one with this? Are they going to agree on a decision? If not, the default mode has to be hashseed, at least for a while, and announce the change of the default value in advance (and perhaps place it behind a feature flag).

Reply:
My thinking was that users who have explicitly set a `hash_seed` should get what they asked for indefinitely, so this default only applies when the component is not configured with a hash seed. I imagine that users who have not configured a hash seed are tolerant of a change of behavior because it would be unwise to use an unconfigured hash_seed. However, you may disagree, and maybe a feature flag is necessary.

The reason why I think the default should change, is that the proportional mode has the same observable probability as hash_seed (always) and does not break or reject data that happens to arrive with OTEP-235 compatible threshold information. The hash_seed mode can't deal with previously OTEP-235 sampled data.

Reply:
> the proportional mode has the same observable probability as hash_seed (always) 

This means that two Collectors, one with v0.102.0 and without an explicit seed will yield the same decision for the same trace ID as a server on v0.103.0, which would default to the proportional mode? If that's the case, I think we are fine and the change is effectively an implementation detail.

If the decisions will be different, I think we should follow the regular deprecation policy.

Reply:
I reverted this aspect of the change. The default can remain `hash_seed` for now. 

I still think a feature-flag and change of default will make sense one day. Users who have this sort of multi-level sampling scheme are, I think, better served by a hierarchy of Proportional and/or Equalizing samplers.

Working on this component has made me realize that despite progress on TraceID-based sampling, OpenTelemetry could use a specification for attribute-based sampling based on the work done in this component, e.g., "When sampling by attribute value, samplers should: (a) apply any hash function they prefer, (b) calculate 56 bits of randomness, (c) express sampling probability using OTEP 235 threshold information in the item, (d) encode OTEP 235 explicit randomness value in the item."

Reply:
> Does that sound right?

Yes, that's correct. The idea is that we give users a period to migrate their collectors to a version that is behaving like a new version, so that a rolling update won't cause disruption on their end.